### ISSUE TITLE
[Bug]  Training HiFiGan (avg loss not decreasing)

### ISSUE THREAD

**Issue by upskyy** [coqui-ai/TTS#1368]
### To Reproduce

```bash
$ CUDA_VISIBLE_DEVICES=0 python3 TTS/recipes/ljspeech/hifigan/train_hifigan.py
```

```python
# train_hifigan.py
import os

from trainer import Trainer, TrainerArgs

from TTS.utils.audio import AudioProcessor
from TTS.vocoder.configs import HifiganConfig
from TTS.vocoder.datasets.preprocess import load_wav_data
from TTS.vocoder.models.gan import GAN

output_path = os.path.dirname(os.path.abspath(__file__))

config = HifiganConfig(
    batch_size=64,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=5,
    epochs=1000,
    seq_len=8192,
    pad_short=2000,
    use_noise_augment=True,
    eval_split_size=10,
    print_step=25,
    print_eval=False,
    mixed_precision=False,
    lr_gen=1e-4,
    lr_disc=1e-4,
    data_path=os.path.join(output_path, "../LJSpeech-1.1/wavs/"),
    output_path=output_path,
)

# init audio processor
ap = AudioProcessor(**config.audio.to_dict())

# load training samples
eval_samples, train_samples = load_wav_data(config.data_path, config.eval_split_size)

# init model
model = GAN(config, ap)

# init the trainer and 🚀
trainer = Trainer(
    TrainerArgs(),
    config,
    output_path,
    model=model,
    train_samples=train_samples,
    eval_samples=eval_samples,
)
trainer.fit()
```

### Expected behavior

I ran ljspeech training with the modified code, but the loss does not drop.
Related issues  ( #1003 #1366 )
Is there anything I can fix?

<img width="1415" alt="스크린샷 2022-03-10 오후 9 41 53" src="https://user-images.githubusercontent.com/54731898/157663632-dce763ff-e941-4707-a2db-b04a3b1ff56c.png">

### Environment

- 🐸TTS Version (e.g., 1.3.0):  0.6.1
- PyTorch Version (e.g., 1.8) :  1.10.2
- Python version:  3.8.12
- OS (e.g., Linux):  Linux 
- CUDA/cuDNN version:  11.3.1
- GPU models and configuration: GeForce 3090
- How you installed PyTorch (`conda`, `pip`, source): conda
- Any other relevant information:

**Comment by erogol** [comment]
What is important is Avg_G_loss and it is going down. 2.2k steps are too early to worry.

I close this, reopen if it does not produce intelligible results after 100k steps,

**Comment by upskyy** [comment]
Even though it exceeds 200k, it still sounds like a mechanical sound. Do I need to edit the hyperparameters separately?

<img width="1411" alt="스크린샷 2022-03-14 오전 11 21 23" src="https://user-images.githubusercontent.com/97871105/158093643-6efa95e6-b62a-49c3-8da6-1ebe54d542e9.png">

**Comment by Aymen08** [comment]
I have same issue.
Even I have a plateau in the average loss, The sound produced is not good.
![EvalTestHifiGAN](https://user-images.githubusercontent.com/57662916/160099210-cfc81dab-f77b-4af2-8aa5-aa442aa3e1f1.png)

**Comment by BoneGoat** [comment]
Here is my attempt with HiFiGAN. Robotic voice after 350k steps and not improving. Used the same config as pretrained HiFiGAN which produces much better quality on my own FastPitch model even though it wasn't trained on the same data.
![Coqui-0 6 2-HiFiGAN-Tensorboard-scalars](https://user-images.githubusercontent.com/7824170/165905310-26e48faf-cf7e-46ba-bf5f-68af42b8ef8f.png)
![Coqui-0 6 2-HiFiGAN-Tensorboard-images](https://user-images.githubusercontent.com/7824170/165905343-468d7e11-3168-4956-b4d9-e3e1a549e3fe.png)

**Comment by erogol** [comment]
Anyone having problems with HifiGAN try the current dev branch with a fix
